sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/StreamingJoinHelper.scala

```diff
@@ -284,7 +287,8 @@ object StreamingJoinHelper extends PredicateHelper with Logging {
           Seq(negateIfNeeded(castedLit, negate))
         case a @ _ =>
           logWarning(
-            s"Failed to extract state value watermark from condition $exprToCollectFrom due to $a")
+            log"Failed to extract state value watermark from condition " +
+              log"${MDC(JOIN_CONDITION, exprToCollectFrom)} due to ${MDC(JOIN_CONDITION, a)}")
```

Comment:
What is `a` here? It seems that you are using a same key for two variables

Reply:
Sorry about this, I did not know we had to have unique keys for the log keys. Fixed now.